src/ports/linux/pnal_filetools.c

```diff
@@ -0,0 +1,213 @@
+ * www.rt-labs.com
+ * Copyright 2020 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "pnal_filetools.h"
+
+#include "pnal.h"
+#include "pf_file.h"
+#include "osal_log.h"
+#include "options.h"
+
+#include <sys/stat.h>
+
+#include <libgen.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+bool pnal_does_file_exist (const char * filepath)
+{
+   struct stat statbuffer;
+
+   return (stat (filepath, &statbuffer) == 0);
+}
+
+/**
+ * @internal
+ * Get the path to the directory where the main binary is located.
+ *
+ * @param path             Out:   Path to directory. Terminated string.
+ * @param size             In :   Size of the stringbuffer. Should be large
+ *                                enough for the full path to the binary.
+ *
+ * @return 0 on success.
+ *         -1 if an error occured.
+ */
+static int pnal_get_directory_of_binary (char * path, size_t size)
+{
+   ssize_t num_bytes = 0;
+
+   num_bytes = readlink ("/proc/self/exe", path, size);
+   if (num_bytes == -1)
+   {
+      return -1;
+   }
+
+   /* Verify that we have the full path (size is big enough) */
+   if (!pnal_does_file_exist (path))
+   {
+      return -1;
+   }
+
+   dirname (path);
+
+   return 0;
+}
+
+int pnal_execute_script (int argc, const char * argv[])
+{
+   int ix;
+   char scriptpath[PNET_MAX_DIRECTORYPATH_LENGTH + PNET_MAX_FILENAME_LENGTH] = {
+      0};
+   char directory_for_binary[sizeof (scriptpath)] = {0};
+   char outputcommand[sizeof (scriptpath) + 200] = {0};
+   bool script_found = false;
+   char * envpath = NULL;
+   char * envpath_copy = NULL;
+   char * tokenpointer;
+   bool firstloop = true;
+   char * searchdir = NULL;
+   int status = 0;
+
+   if (argc < 1)
+   {
+      LOG_ERROR (PF_PNAL_LOG, "PNAL(%d): No script name given.\n", __LINE__);
+      return -1;
+   }
+
+   /* Look in the same directory as the main binary */
+   if (
+      pnal_get_directory_of_binary (
+         directory_for_binary,
+         sizeof (directory_for_binary)) != 0)
+   {
+      LOG_ERROR (
+         PF_PNAL_LOG,
+         "PNAL(%d): Failed to locate the directory of main binary.\n",
+         __LINE__);
+      return -1;
+   }
+   pf_file_join_directory_filename (
+      directory_for_binary,
+      argv[0],
+      scriptpath,
+      sizeof (scriptpath));
+   if (pnal_does_file_exist (scriptpath))
+   {
+      script_found = true;
+   }
+
+   /* Look in current directory */
+   if (script_found == false)
+   {
+      pf_file_join_directory_filename (
+         NULL,
+         argv[0],
+         scriptpath,
+         sizeof (scriptpath));
+      if (pnal_does_file_exist (scriptpath))
+      {
+         script_found = true;
+      }
+   }
+
+   /* Search in $PATH. Use a copy not to modify the path  */
```

Comment:
Consider creating a function similar to  "pnal_get_directory_of_binary()" for this block.

Reply:
Implemented` pnal_locate_script()`
Done

Reply:
Now having` pnal_create_searchpath()` instead.